build.go

```diff
@@ -131,37 +147,53 @@ func buildBlock(sb io.StringWriter, parent *Directive, block Directives, depth i
 			directive := Enquote(stmt.Directive)
 			_, _ = sb.WriteString(directive)
 
-			// special handling for if statements
-			if directive == "if" {
-				_, _ = sb.WriteString(" (")
-				for i, arg := range stmt.Args {
-					if i > 0 {
-						_, _ = sb.WriteString(" ")
+			if options.ExternalBuilds != nil {
+				extDirectivesMap := make(map[string]Builder)
```

Comment:
`buildBlock` is called recursively. I don't think you want to rebuild the `extDirectivesMap` map each time. I think it would be better to execute this loop in the `Build` function.